src/System.Numerics.Vectors/tests/CommonHashCodeTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Numerics.Tests
+{
+    public static class CommonHashCodeTests
+    {
+        private const int ElementsCount = 100000;
+
+        private static void CountVector2Collisions(Func<int, Vector2> vectorConstructor, out int collisions, out int maxCollisions)
+        {
+            var coll = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Select(x => new
+                {
+                    Hash = x.Key,
+                    x.ToList().Count
+                }).ToList();
+
+            int collidedHashesCount = coll.Count(x => x.Count > 1);
+
+            collisions = coll.Where(x => x.Count > 1).Sum(x => x.Count) - collidedHashesCount;
+
+            maxCollisions = coll.Max(x => x.Count);
+        }
+
+        private static void CountVectorCollisions<T>(Func<int, Vector<T>> vectorConstructor,
+            out int collisions) where T : struct
+        {
+            var coll = Enumerable.Range(0, ElementsCount)
+                .Select(vectorConstructor)
+                .GroupBy(x => x.GetHashCode())
+                .Select(x => new
+                {
+                    Hash = x.Key,
+                    x.ToList().Count
+                }).ToList();
+
+            int collidedHashesCount = coll.Count(x => x.Count > 1);
+
+            collisions = coll.Where(x => x.Count > 1).Sum(x => x.Count) - collidedHashesCount;
+        }
+
+        [Fact]
+        public static void CombineHashCodesVector2Tests()
+        {
+            int collisions;
+            int maxCollisions;
+
+            // On Highly distrubuted data, percent of collisions has to be less than 1%.
+            CountVector2Collisions(x => new Vector2 {X = x, Y = x + ElementsCount}, out collisions, out maxCollisions);
+            Assert.True(0.01 >= collisions/(float) ElementsCount);
+
+            // Every coordinate not greater than 10, total collisions could be 99.9%, but maximum elements with equal hash < 25%.
+            CountVector2Collisions(x => new Vector2 {X = x%10, Y = (x + ElementsCount)%10}, out collisions, out maxCollisions);
```

Comment:
You are right, but I'm checking here maximum value of distribution density function. Removed next line.